src/error.ts

```diff
@@ -68,12 +68,12 @@ export function createError<DataT = any>(
   input:
     | string
     | (Partial<H3Error<DataT>> & { status?: number; statusText?: string }),
-): H3Error {
+) {
```

Comment:
Do we need to add an explicit return type here to ensure the generic propagates?

Reply:
With the addition of generics to `isError`, this is now inferred correctly as `H3Error<DataT>`.